src/shared/util.js

```diff
@@ -1053,6 +1055,16 @@ function isAscii(str) {
   return /^[\x00-\x7F]*$/.test(str);
 }
 
+function fromUTF16(str) {
```

Comment:
*This applies to the new helper functions below as well.*

Do we need this on both threads? If not, can we perhaps move it to `src/core/core_utils.js` instead?